paper-api/src/main/java/io/papermc/paper/event/player/PlayerCustomClickEvent.java

```diff
@@ -0,0 +1,55 @@
+package io.papermc.paper.event.player;
+
+import io.papermc.paper.dialog.Dialog;
+import io.papermc.paper.dialog.DialogResponse;
+import io.papermc.paper.dialog.ValidatedDialogResponse;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.nbt.api.BinaryTagHolder;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+@ApiStatus.Experimental
+@ApiStatus.NonExtendable
+@NullMarked
+public abstract class PlayerCustomClickEvent extends Event {
```

Comment:
I feel like this should extend `PlayerEvent`.

Reply:
No, it won't. It will not have a Player always, as you can use this in the configuration phase. Once Owen's pr for configuration phase API is in, it will have a connection instance, one implementation of which will have a player, the other just a player profile.

Reply:
Ah, I see, didn't think of that possibility, makes sense.